src/thread_pool.rs

```diff
@@ -259,7 +259,7 @@ impl ThreadPool {
     {
         // We cannot shrink the pool from within the pool, so we simply refuse
         // the request and return the same size.
-        if !WorkerThread::current().is_null() {
+        if WorkerThread::with(|worker_thread| worker_thread.is_some()) {
```

Comment:
Nit: If we are going to do it like this maybe it should just be an associated function `WorkerThread::is_worker()` or something?